### PR TITLE
Add new documents

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. how to add TASH command
2. how to add (port) new board
3. modify information of Tizen RT
4. add API reference

### CODE REVIEW

**Comment by sunghan-chang** [comment]
This is an initial draft.

DONT MERGE THIS.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by jsdosa** [comment]
@sunghan-chang  i have one question. what is port of second description?

**Comment by sunghan-chang** [comment]
@jsdosa That is not done. I need to fill it more. build/configs will be added at second.

**Comment by sunghan-chang** [comment]
When you check this changes, please access [[my repo]](https://github.com/sunghan-chang/TizenRT/tree/readme/docs). You had better see it through markdown view.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@btheosam @samirsahu1947 @shivgarg Please review this. To see it, please visit [[here]](https://github.com/sunghan-chang/TizenRT/tree/readme/docs).

**Comment by seinfra** [comment]
Code Rule Check OK.

File: docs/HowToAddnewBoard.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# How to add(port) a new board
+
+To support a new board, two big parts, [arch](#arch) and [configs](#configs), should be modified.
+
+## Arch
+
+This part contains BSP (Board Support Package) for chip and board where are *os/arch* folder.
```
Comment (shivgarg) [inline docs/HowToAddnewBoard.md]: This line needs to restructured. For eg.  This part contains BSP(Board Support Package) for chip and board and chip. The path to this part is *os/arch*.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by btheosam** [comment]
The new board addition readme should be refined properly.

File: docs/HowToAddnewBoard.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+1. Add a new folder with board name.
+```
+mkdir build/configs/<board-name>
+```
+
+2. Add the *script* folder and add the linker script file.
+```
+mkdir build/configs/<board-name>/scripts
+```
+
```
Comment (samirsahu1947) [inline docs/HowToAddnewBoard.md]: We can add an exmaple for adding simple configuration for testing with hello application + Tash shell

File: docs/HowToAddnewBoard.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+### defconfig
+
+This is a configurations set of representing folder name.  
+This is moved to .config at executing configure.sh and used to build Tizen RT.
+
+### Make.defs
+
+This is a build options set like setting of including path, ARM build options, linker script name and so on.  
+There is the DOWNLOAD definition to support "make download xx" command too.
+
```
Comment (samirsahu1947) [inline docs/HowToAddnewBoard.md]: We can add general guidelines for how application should call BSP code without invoking directly invoking

File: docs/HowToAddnewBoard.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# How to add(port) a new board
+
+Changes in *[arch](#arch)* and *[configs](#configs)* folders are mandantory to support a new board.  
+Adding static library or expanding driver will be provided later.
+
```
Comment (samirsahu1947) [inline docs/HowToAddnewBoard.md]: Addition of New Drivers into os/driver model ,
Getting the Wi-Fi driver up and hooking with the wpa_supplicant can be a part of separate doc linked to this for refrence

File: docs/HowToAddnewBoard.md (hunk 3 above)
Comment (samirsahu1947) [inline docs/HowToAddnewBoard.md]: We have to add a section of common issues faced and preferred guidelines like config entry in Kconfig , following System API structure , Usage of correct driver interface etc can be added here

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**